rootfs/api/models/release.py

```diff
@@ -148,6 +148,27 @@ def rollback(self, user, version):
                 new_release.delete()
             raise
 
+    def delete(self, *args , **kwargs):
+        """Delete release DB record and any RCs from the affect release"""
+        try:
+            labels = {
+                'app': self.app.id,
+                'version': self.version
+            }
+            controllers = self._scheduler.get_rcs(self.app.id, labels=labels)
+            for controller in controllers.json()['items']:
+                self._scheduler._delete_rc(self.app.id, controller['metadata']['name'])
+        except KubeHTTPException as e:
+            # 404 means they were already cleaned up
+            if e.status_code is not 404:
+                # Another problem came up
+                message = 'Could not to cleanup RCs for release {}'.format(self.version)
+                log_event(self.app, message)
+                logger.warning(message + ' - ' + str(e))
+
+        super(Release, self).delete(*args, **kwargs)
```

Comment:
The codebase does not have many (any?) `finally` but it feels like a good practice to do since we want to delete no matter what, added it in